File: src/test/java/base/BaseTest.java
package base;

import config.KobitonConfig;
import io.appium.java_client.android.AndroidDriver;
import org.openqa.selenium.remote.DesiredCapabilities;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;

import java.net.MalformedURLException;
import java.net.URISyntaxException;
import java.net.URL;

public class BaseTest {

    protected AndroidDriver driver = null;

    @BeforeClass
    protected void init() throws MalformedURLException, URISyntaxException {
        URL serverURL = KobitonConfig.getServerURL();
        DesiredCapabilities capabilities = KobitonConfig.getAndroidDesiredCapabilities();
        driver = new AndroidDriver(serverURL, capabilities);
    }

    @AfterClass
    protected void tearDown() {
        driver.quit();
    }

}

File: src/test/kotlin/base/BaseTest.kt
package base

import io.appium.java_client.service.local.AppiumDriverLocalService
import io.appium.java_client.service.local.AppiumServiceBuilder
import org.testng.annotations.AfterClass
import org.testng.annotations.BeforeTest
import java.io.File

open class BaseTest {

    protected var service: AppiumDriverLocalService? = null

    @BeforeTest
    open fun setupAppium() {
        service = AppiumServiceBuilder().apply {
            withAppiumJS(File("//usr//local//lib//node_modules//appium/build//lib//main.js"))
            withIPAddress("127.0.0.1")
            usingPort(4723)
        }.build()
    }

    @AfterClass
    open fun tearDownAppium() {
        service?.stop()
    }

}